maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -72,6 +75,36 @@ sealed class CommandOutput {
     data class AIDefects(val defects: List<Defect>, val screenshot: Buffer) : CommandOutput()
 }
 
+interface FlowController {
+    suspend fun waitIfPaused()
+    fun pause()
+    fun resume()
+    val isPaused: Boolean
+}
+
+class DefaultFlowController : FlowController {
+    private var _isPaused = false
+    
+    override suspend fun waitIfPaused() {
+        while (_isPaused) {
+            if (!coroutineContext.isActive) {
+                break
+            }
+            Thread.sleep(500)
```

Comment:
The right way to suspend coroutine is delay, I don't think this should be needed